### ISSUE TITLE
Has ProgLoss included in YOLOv26 been implemented?

### ISSUE THREAD

**Issue by donny95** [ultralytics/ultralytics#23239]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello,

According to the paper, starting from YOLOv26, DFL Loss is no longer used and ProgLoss is adopted instead.
However, in the YOLOv26 training logs that I am currently running, dfl_loss is still being computed.

Is there a planned update regarding this behavior?
Or is there a specific configuration or setting that needs to be changed to enable ProgLoss?

Thank you.

<img width="1401" height="130" alt="Image" src="https://github.com/user-attachments/assets/56cff122-49d0-4ae9-bab1-e73a6c79e682" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @donny95, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report (e.g., you believe the YOLO26 implementation should not be reporting `dfl_loss` in your run), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In particular, please include:
-Exact `pip show ultralytics` output
-Your full training command (CLI) or script (Python)
-The model file/name you’re using and any overrides
-A short snippet of the training log showing the reported losses (your screenshot helps 👍, but text logs are best)
-OS, Python, PyTorch, and GPU/CUDA details

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The `dfl_loss` here is just L1 loss. It's just using old metric label.

https://github.com/ultralytics/ultralytics/blob/fbb0caf580f4447a87b6b537f67ab9625412c70e/ultralytics/utils/loss.py#L149

**Comment by donny95** [comment]
> The `dfl_loss` here is just L1 loss. It's just using old metric label.
> 
> [ultralytics/ultralytics/utils/loss.py](https://github.com/ultralytics/ultralytics/blob/fbb0caf580f4447a87b6b537f67ab9625412c70e/ultralytics/utils/loss.py#L149)
> 
> Line 149 in [fbb0caf](/ultralytics/ultralytics/commit/fbb0caf580f4447a87b6b537f67ab9625412c70e)
> 
>  F.l1_loss(pred_dist[fg_mask], target_ltrb[fg_mask], reduction="none").mean(-1, keepdim=True) * weight



 yolo train  model=./yolo26x.pt   data=/workspace/ultralytics/custom/custom.yaml   epochs=200   imgsz=640   device=1,2,3,4  batch=128   workers=4   optimizer=MuSGD   lr0=0.0001 momentum=0.9 

I am currently training on a custom dataset using this command.
I have also updated the Ultralytics library to the latest version.

During training, it still appears that dfl_loss is being computed.
Do you mean that only the name dfl_loss is being used, and that the actual loss function is ProgLoss?

**Comment by glenn-jocher** [comment]
Yes—on Ultralytics YOLO26 the DFL *module* is removed (so `reg_max` falls back and `DFLoss` isn’t constructed), but the training logger still reports the bbox distance term under the legacy `dfl_loss` name for continuity; ProgLoss is a training-time balancing strategy and won’t appear as a separate “ProgLoss” scalar in logs (see the YOLO26 overview in the [Ultralytics YOLO26 docs](https://docs.ultralytics.com/models/yolo26/) and the `BboxLoss` `reg_max > 1 else None` path in the [loss implementation reference](https://docs.ultralytics.com/reference/utils/loss/)).  

```bash
yolo checks
```
